frontend/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -0,0 +1,375 @@
+/**
+ * @license
+ * Copyright 2018-2022 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement, useState } from "react"
+import {
+  DataEditor as GlideDataEditor,
+  GridCell,
+  GridColumn,
+  DataEditorProps,
+  DataEditorRef,
+  GridSelection,
+  CompactSelection,
+  GridMouseEventArgs,
+} from "@glideapps/glide-data-grid"
+import { useColumnSort } from "@glideapps/glide-data-grid-source"
+
+import withFullScreenWrapper from "src/hocs/withFullScreenWrapper"
+import { Quiver } from "src/lib/Quiver"
+import { logError } from "src/lib/log"
+
+import {
+  getCellTemplate,
+  fillCellTemplate,
+  getColumnSortMode,
+  determineColumnType,
+  ColumnType,
+} from "./DataFrameCells"
+import ThemedDataFrameContainer from "./DataFrameContainer"
+
+const ROW_HEIGHT = 35
+const MIN_COLUMN_WIDTH = 35
```

Comment:
Is there a reason why these numbers are chosen out of curiosity?

Reply:
They are the current defaults of glide-data-grid. Since we cannot easily access the defaults from within the library, I added it here. Also, this allows us to keep the UI more stable even when glide-data-grid updates its defaults.